web/frontend/src/hooks/robot-client.ts

```diff
@@ -25,28 +25,39 @@ const context = {
 
 export const useRobotClient = () => context;
 
+export type DisconnectCallback = () => void
+export type ConnectCallback = () => DisconnectCallback | undefined
+
 /**
- * This hook will fire whenever a connection occurs.
+ * Pass a callback to this hook that will fire whenever an initial connection or reconnect occurs.
+ * 
+ * The callback can return a disconnection callback that will fire whenever a disconnect or unmount occurs.
+ * 
+ * @example
+ * ```ts
+ * useConnect(() => {
+ *    const clearPoll = startPolling()
+ *    return () => clearPoll()
+ * })
+ * ```
  */
-export const useConnect = (callback: () => void) => {
+export const useConnect = (callback: ConnectCallback) => {
   const { connectionStatus } = useRobotClient();
 
-  const unsub = connectionStatus.subscribe((value) => {
-    if (value === 'connected') {
-      callback();
-    }
-  });
-
-  onDestroy(() => unsub());
-};
-
-/**
- * This hook will fire whenever a disconnect occurs or when a component unmounts.
- */
-export const useDisconnect = (callback: () => void) => {
-  const { statusStream } = useRobotClient();
+  let disconnectCallback: DisconnectCallback | undefined
 
-  statusStream.subscribe((update) => update?.on('end', callback));
+  onMount(() => {
+    const unsubscribe = connectionStatus.subscribe((value) => {
+      if (value === 'connected') {
+        disconnectCallback = callback();
+      } else if (value === 'reconnecting') {
```

Comment:
Do we want to call the `disconnectCallback` for all non-`connected` disconnect status events? Or just `reconnecting`?

Reply:
I think that we could punt on this and solve by improving the state machine. Right now as far as I understand if a disconnect occurs it transitions from `connected` to `reconnecting` immediately. I think that we could potentially introduce problems now if we triggered this callback during any other states.